packages/components/src/FilterDatePicker/FilterDatePicker.spec.tsx

```diff
@@ -86,81 +86,136 @@ describe("<FilterDatePicker />", () => {
     })
   })
 
-  it("closes the popover when a valid date has been submitted via the text input field", async () => {
-    const { getByRole, getByLabelText } = render(<FilterDatePickerWrapper />)
-    const triggerButton = getByRole("button", {
-      name: "Drank",
-    })
+  describe("Text input", () => {
+    it("validates the date on blur", async () => {
+      const { getByRole, getByLabelText, getByText } = render(
+        <FilterDatePickerWrapper />
+      )
+      const triggerButton = getByRole("button", {
+        name: "Drank",
+      })
 
-    await user.click(triggerButton)
-    const dialog = getByRole("dialog")
+      await user.click(triggerButton)
+      const dialog = getByRole("dialog")
 
-    await waitFor(() => {
-      expect(dialog).toBeInTheDocument()
+      await waitFor(() => {
+        expect(dialog).toBeInTheDocument()
+      })
+
+      const inputDate = getByLabelText("Date")
+      await user.clear(inputDate)
+      await user.type(inputDate, "32/13/2022")
+      await user.tab()
+
+      await waitFor(() => {
+        expect(getByText("32/13/2022 is an invalid date")).toBeInTheDocument()
+      })
     })
 
-    const inputDate = getByLabelText("Date")
-    await user.clear(inputDate)
-    await user.type(inputDate, "07/06/2022")
-    await user.tab()
+    describe("Pressing Enter in the text input", () => {
```

Comment:
Was this a new bug you found and fixed?

Reply:
Yes and no. You had tests for it, but because the "on submit" was in the wrong place, the test was inaccurate. Moreso that I split it.